schism/compiler.ss

```diff
@@ -709,12 +731,16 @@
             (tc (caddr t))
             (ta (cadddr t)))
         (and (eq? 'bool (car tc)) (eq? 'bool (car ta))
-             (not (eq? (cadr tc) (cadr ta))))
-        (if (cadr tc)
-            ;; (if (if TT #t #f) C A) -> (if TT C A)
-            (simplify-conditional tt c a)
-            ;; (if (if TT #f #t) C A) -> (if TT A C)
-            (simplify-conditional tt a c))))
+             (cond
+              ((eq? (cadr tc) (cadr ta))
```

Comment:
Should this be `equal?` instead? Is there a case where the same (as in, same memory location) expression gets used in both the consequent and alternative?

Reply:
In this case tc and tf are either `(bool #t)` or `(bool #f)` -- i.e. the cadr is a raw boolean, so we can use eq?.

Reply:
:+1: